src/main/scala/org/deeplearning4j/scalphagozero/scoring/GameResult.scala

```diff
@@ -34,15 +57,35 @@ object GameResult {
   /**
     * Compute the game result from the current state.
     *
-    * @param gameState GameState instance
+    * @param goBoard GoBoard instance
     * @return GameResult object
     */
-  def computeGameResult(gameState: GameState): GameResult = {
-    val territory = evaluateTerritory(gameState.board)
-    new GameResult(
-      territory.numBlackTerritory + territory.numBlackStones,
-      territory.numWhiteStones + territory.numWhiteStones,
-      7.5
+  def computeGameResult(goBoard: GoBoard): GameResult = {
+    val territoryMap = evaluateTerritory(goBoard)
+
+    var numBlackStones = 0
+    var numWhiteStones = 0
+    var numBlackTerritory = 0
+    var numWhiteTerritory = 0
+    var numDame = 0
+
+    for ((_, status) <- territoryMap) {
```

Comment:
FYI @guizmaii the best way here would be to import cats: `territoryMap.values.foldMap(t => Map(t -> 1))` (it uses the `Int` `Monoid` and return a `Map[Type, Int]`), and probably use cats in more places.

Reply:
@guizmaii @sderosiaux let's leave cats out, please. You might see this differently, but it's often the beginning of the end of a project's readability :)

Reply:
Sure @maxpumperla, no worries.

Reply:
@sderosiaux thanks ! I didn't know that :) but as requested. I'll not add Cats ;)